components/match2/commons/recent_matches.lua

```diff
@@ -0,0 +1,245 @@
+	local limit = tonumber(args.limit or 20) or 20
+
+	local data = RecentMatches._getData(conditions, limit)
+
+	if not data then
+		return mw.html.create('div')
+			:addClass('text-center')
+			:wikitext('<br/>No Recent Results<br/><br/>')
+	end
+
+	local display = ''
+
+	for _, item in ipairs(data) do
+		display = display .. RecentMatches._row(item)
+	end
+
+	return display
+end
+
+-- overridable functions
+function RecentMatches.requireOpponentModules()
+	return Lua.import('Module:OpponentDisplay', {requireDevIfEnabled = true}),
+		Lua.import('Module:Opponent', {requireDevIfEnabled = true})
+end
+
+function RecentMatches.buildConditions(args)
+	local featured = args.featured == 'true'
+
+	local conditions = '[[dateexact::1]] AND [[finished::1]] AND [[date::<' .. _CURRENT_DATE_STAMP .. ']]'
+	if featured then
+		conditions = conditions .. ' AND [[publishertier::>]]'
+	end
+
+	return conditions
+end
+
+function RecentMatches.scoreDisplay(opponentLeft, opponentRight, winner)
+	local leftScore = RecentMatches.getOpponentScore(opponentLeft)
+	local rightScore = RecentMatches.getOpponentScore(opponentRight)
+
+	local scoreDisplay = RecentMatches._displayOpponentScore(leftScore, winner == 1)
+		.. ':'
+		.. RecentMatches._displayOpponentScore(rightScore, winner == 2)
+
+	return scoreDisplay
+end
+
+function RecentMatches.getOpponentScore(opponent)
+	local score
+	if opponent.status == _SCORE_STATUS then
+		score = opponent.score
+	else
+		score = opponent.status or ''
+	end
+
+	return score
+end
+
+-- helper functions
+function RecentMatches._displayOpponentScore(score, isWinner)
+	return (isWinner and '<b>' or '')
+		.. score
+		.. (isWinner and '</b>' or '')
+end
+
+function RecentMatches._getData(conditions, limit)
+	local data = mw.ext.LiquipediaDB.lpdb('match2', {
+		conditions = conditions,
+		order = 'date desc, liquipediatier asc, tournament asc',
+		query = 'match2opponents, winner, resulttype, pagename, tournament, '
+			.. 'tickername, icon, date, publishertier, vod, extradata, parent',
+			--readd icondark to the query once it is fixed in lpdb
+		limit = limit
+	})
+
+	if type(data[1]) == 'table' then
+		return data
+	end
+	mw.logObject(data)
+end
+
+function RecentMatches._getTableClass(winner, publishertier)
+	local class = 'wikitable wikitable-striped infobox_matches_content recent-matches-'
+
+	if winner == 1 then
+		class = class .. 'left'
+	elseif winner == 2 then
+		class = class .. 'right'
+	else
+		class = class .. 'draw'
+	end
+
+	if String.isNotEmpty(publishertier) then
+		class = class .. '-publishertier'
+	end
+
+	return class
+end
+
+function RecentMatches._checkForInelligableOpponent(opponent)
+	local name = string.lower(opponent.name or '')
+	local template = string.lower(opponent.template or '')
+
+	return name == 'bye'
+		or name == 'tba'
+		or name == 'tbd'
+		or template == 'bye'
+		or template == 'tba'
+		or template == 'tbd'
+		or (String.isEmpty(name) and String.isEmpty(template))
```

Comment:
Add a list at the the top and do Table.contains. Alternatively make a Set and do Set.contains.